### PR TITLE
Append error msg in cross signing

### PR DESCRIPTION
**Pull request by bitfriend**
Append error msg for the missed cancel codes

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/verification/mod.rs [hunk 1]
```diff
@@ -364,6 +364,10 @@ impl Cancelled {
             }
             CancelCode::InvalidMessage => "The received message was invalid.",
             CancelCode::KeyMismatch => "The expected key did not match the verified one",
+            CancelCode::MismatchedCommitment => {
+                "The device receiving this error can ignore the verification request."
```
Comment (poljar) [inline crates/matrix-sdk-crypto/src/verification/mod.rs:368]: That's not the reason from the spec: https://spec.matrix.org/unstable/client-server-api/#verification-messages-specific-to-sas

**Review by poljar — CHANGES_REQUESTED** [review]
Looks good, except one of the reasons doesn't match to the reason from the spec.

File: crates/matrix-sdk-crypto/src/verification/mod.rs (hunk 1 above)
Comment (bitfriend) [reply crates/matrix-sdk-crypto/src/verification/mod.rs:368]: This error msg was copied from [ruma](https://github.com/ruma/ruma/blob/31331f3165fd8eba44ca2d7e7873e4083c6adac1/crates/ruma-events/src/key/verification/cancel.rs#L118).

Comment (poljar) [reply crates/matrix-sdk-crypto/src/verification/mod.rs:368]: Well the docstring can differ from the `reason`, I would expect the docstring to explain things in more detail. The `reason` string on the other hand should match to the one in the spec.

Comment (bitfriend) [reply crates/matrix-sdk-crypto/src/verification/mod.rs]: Sorry, my mistake closed this PR.
Will fix it in [another PR](https://github.com/matrix-org/matrix-rust-sdk/pull/3217).

**Pull request closed without merging**